packages/mdc-textfield/mdc-textfield.scss

```diff
@@ -123,10 +124,197 @@ $mdc-textfield-disabled-border-on-dark: rgba(white, .3);
     transform: translateY(-100%) scale(.75, .75);
     cursor: auto;
   }
-  // stylelint-enable plugin/selector-bem-pattern  
+  // stylelint-enable plugin/selector-bem-pattern
 }
 
-.mdc-textfield--upgraded:not(.mdc-textfield--fullwidth) {
+.mdc-textfield--box {
+  display: inline-flex;
+  position: relative;
+  height: 56px;
+  border-radius: 4px 4px 0 0;
+  background-color: rgba(black, .04);
+  overflow: hidden;
+
+  @include mdc-theme-dark(".mdc-textfield", true) {
+    background-color: rgba(white, .1);
+  }
+
+  &::before,
+  &::after {
+    position: absolute;
+    top: 0;
+    left: 0;
+    // TODO ripple positioning logic?
+    width: 100%;
+    height: 100%;
+    border-radius: 50%;
+    background-color: inherit;
+    content: "";
+    opacity: 0;
+    pointer-events: none;
+  }
+
+  // NOTE: For some reasons, stylelint complains that the two patterns below don't follow BEM.
+  // However, it seems to emit the wrong selector for that pattern. Thus, we disable it above where
+  // we would normally disable it (&.mdc-textfield--disabled) as a workaround.
+
+  // stylelint-disable plugin/selector-bem-pattern
+  // TODO: Refactor this out once #194 lands
+  $opacity-duration: 83ms;
+
+  &::before {
+    transition: opacity 17ms linear, transform 0ms $opacity-duration+2 linear;
+  }
+
+  &:hover {
+    &::before {
+      transition:
+        opacity $opacity-duration linear 17ms,
+        transform 250ms $mdc-animation-linear-out-slow-in-timing-function 17ms;
+      opacity: 1;
+    }
+  }
+
+  .mdc-textfield__input {
+    align-self: flex-end;
+    height: 100%;
+    box-sizing: border-box;
+    padding: 20px 16px 0;
+  }
+
+  .mdc-textfield__label {
+    @include mdc-rtl-reflexive-position(left, 16px);
+
+    position: absolute;
+    bottom: 20px;
+    color: rgba(black, .6);
+    pointer-events: none;
+    // Force the label into its own layer to prevent to prevent visible layer promotion adjustments
+    // when the ripple is activated behind it.
+    will-change: transform;
+
+    @include mdc-theme-dark(".mdc-textfield") {
+      @include mdc-theme-prop(color, text-secondary-on-dark);
+    }
+
+    &--float-above {
+      transform: translateY(-50%) scale(.75, .75);
+    }
+  }
+
+  &.mdc-ripple-upgraded {
+    @include mdc-ripple-base;
+    // TODO: Add ::before as ripple bg once #194 lands
```

Comment:
Curious. I had initially slowed it down but still couldn't see it animating out. I'm thinking the darker background of the text box background is the same color as the ink wash, which makes it appear as though there is no fade out. All the user can see is the foreground ripple (which doesn't animate out afaik)

Either way, if motion signed off, that's good enough for me. 👍